gui/wxpython/startup/utils.py

```diff
@@ -87,3 +88,72 @@ def create_database_directory():
         pass
 
     return None
+
+
+def _get_startup_location_in_distribution():
+    """Check for startup location directory in distribution.
+
+    Returns startup location if found or None if nothing was found.
+    """
+    gisbase = os.getenv("GISBASE")
+    startup_location = os.path.join(gisbase, "demolocation")
+
+    # Find out if startup location exists
+    if os.path.exists(startup_location):
+        return startup_location
+    return None
+
+
+def _copy_startup_location(grassdatabase, startup_location):
+    """Copy the simple startup_location with some data to GRASS database.
+
+    Returns True if successfully copied or False when an error was encountered.
+    """
+    src = startup_location
+    dst = os.path.join(grassdatabase, "demolocation")
+
+    # Copy source startup location into GRASS database
+    try:
+        copytree(src, dst, ignore=ignore_patterns('*.tmpl','Makefile*'))
+        return True
+    except (IOError, OSError):
+        pass
+    return False
+
+
+def _create_startup_mapset(grassdatabase, startup_location):
+    """Create the new empty startup mapset named after user.
+
+    Returns True if successfully created or False when an error was encountered.
+    """
+    name = get_default_mapset_name()
+    mapset_path = os.path.join(grassdatabase, startup_location, name)
+    counter = 1
+
+    while os.path.exists(mapset_path):
+        name = name + "_" + str(counter)
```

Comment:
You are actually modifying the name variable here. If you run this more than once, you will get username, username_1, username_1_1, username_1_1_1, ... while you probably wanted username, username_2, username_3, ...

Since this won't be used for the first time user, you can test this before fixing the modification of the location in the distribution. It is sufficient just to delete the config and grassdata dir as in case of a single test (i.e., just do the single test multiple times.)